### PR TITLE
Metamorph: clean up instrument graph for datasets with multiple series

### PR DESCRIPTION
**Pull request by melissalinkert**
Backported from private PR.  For Metamorph datasets with multiple series, this should reduce size of the generated OME-XML/OMERO object graph by using a single ```Instrument``` for every ```Image```, instead of one ```Instrument``` per ```Image```.

I tested primarily with ```data_repo/curated/hms/HCT116.nd```, which has 32 series.  Without this PR, ```showinf -nopix -omexml HCT116.nd``` should show 32 ```Instrument```s, only the first of which has a ```Detector```.  Each ```Image``` should be linked to the ```Instrument``` with the same index (so only the first ```Image``` has a valid linked ```Detector```).

With this PR, the same test should show a single ```Instrument``` with a single ```Detector```, which is linked to every ```Image```.

It would also be good to perform the same pair of tests with ```data_repo/curated/qa-16907``` to verify that the changes from https://github.com/openmicroscopy/bioformats/pull/2698 haven't been broken by this.

### CODE REVIEW

File: components/formats-gpl/src/loci/formats/in/MetamorphReader.java [hunk 1]
```diff
@@ -871,12 +874,13 @@ else if (firstSeriesChannels != null) {
 
           if ((int) wave[waveIndex] >= 1) {
             // link LightSource to Image
+            int laserIndex = i * getEffectiveSizeC() + c;
```
Comment (sbesson) [inline components/formats-gpl/src/loci/formats/in/MetamorphReader.java]: As there is now only instrument/detector per fileset, should the DetectorSettings object setters above be updated in a similar manner by creating a `detectorSettingsIndex` using the series/channel and calling `setDetectorSettingsxxx(value, 0, settingsIndex)`?

Comment (melissalinkert) [inline components/formats-gpl/src/loci/formats/in/MetamorphReader.java]: No, I don't think so.  The indexes on the ```setDetectorSettings*``` methods are for ```Image``` and ```Channel```, not ```Instrument```, so I think what is here will continue to work as expected.

Comment (sbesson) [inline components/formats-gpl/src/loci/formats/in/MetamorphReader.java]: You're right (<a href="http://static.javadoc.io/org.openmicroscopy/ome-xml/5.4.0/ome/xml/meta/MetadataStore.html#setDetectorSettingsBinning-ome.xml.model.enums.Binning-int-int-">setDetectorSettingsBinning</a> and <a href="http://static.javadoc.io/org.openmicroscopy/ome-xml/5.4.0/ome/xml/meta/MetadataStore.html#setDetectorSettingsReadOutRate-ome.units.quantity.Frequency-int-int-">setDetectorSettingsReadOutRate</a> for external references). No need to change these callss

**Comment by mtbc** [comment]
Couldn't find `data_repo/curated/qa-16907` so using `apache_repo/16907/`.

**Comment by mtbc** [comment]
Outwith this PR but in metadata we have `showinf` show things like,
```
X position for position #1: ome.units.quantity.Length: value[-5276.3], unit[reference frame] stored as java.lang.Double
```
maybe these could be made prettier.

**Comment by mtbc** [comment]
For the QA-16907 data we have the expected single,
```xml
  <Instrument ID="Instrument:0">
    <Detector ID="Detector:0:0" Type="Other"/>
    <Objective ID="Objective:0:0" LensNA="0.2"/>
  </Instrument>
```
and the image has,
```
    <InstrumentRef ID="Instrument:0"/>
    <ObjectiveSettings ID="Objective:0:0" RefractiveIndex="1.0"/>
```
so we don't seem to have broken #2698.

**Comment by mtbc** [comment]
Using `data_repo/curated/metamorph/hms/HCT116.nd` (I guess the directories got rearranged a bit) I do indeed see the expected single instrument,
```xml
  <Instrument ID="Instrument:0">
    <Detector ID="Detector:0:0" Type="Other"/>
  </Instrument>
```
and each of the 32 images indeed has,
```xml
    <InstrumentRef ID="Instrument:0"/>
```

**Comment by mtbc** [comment]
https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-merge-repository-subset/387/ looks good. **Good to merge.**

**Pull request merged**